### PR TITLE
Emoncms API now provides a Unit of Measurement

### PR DESCRIPTION
**Pull request by borpin**
## Proposed change

Update to add a variable to use the Unit of measurment included in the Emoncms API call.

Minor other enhancements.

As a user is required to add a variable to make the change this should be regarded as non-breaking.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/home-assistant/pull/32042
- This PR fixes or closes issue: https://github.com/home-assistant/home-assistant/issues/31251

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by borpin** [comment]
I note that the Lint Text check has failed. I cannot see where 'emoncms' has been used in place of 'Emoncms' within the text.

File: source/_integrations/emoncms.markdown [hunk 1]
```diff
@@ -21,17 +21,20 @@ sensor:
   id: 1
 ```
 
+As of Feb 2020 including the option `use_emoncms_unit` will discover all sensors from Emoncms and use the unit of measurement specified in the Feed in Emoncms.  Tested with [Emoncms](https://github.com/emoncms/emoncms) V10.1.13.
+
 ## Configuration variables
```
Comment (klaasnicolaas) [inline source/_integrations/emoncms.markdown:26]: I see that the way in which the config variables are described here is totally inconsistent with how we handle it today. I assign myself to this PR to fix this.

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

File: source/_integrations/emoncms.markdown [hunk 2]
```diff
@@ -21,17 +21,54 @@ sensor:
...
-- `value_template` (*Optional*): Defines a [template](/docs/configuration/templating/#processing-incoming-data) to alter the feed value.
-- `scan_interval` (*Optional*): Defines the update interval of the sensor in seconds.
-- `unit_of_measurement` (*Optional*): Defines the unit of measurement of for all the sensors. default is "W".
+{% configuration %}
+api_key:
+  description: The read API key for your Emoncms user.
+  required: true
+  type: string
+url:
+  description: "The base URL of Emoncms, use <https://emoncms.org> for the cloud based version. For self hosted Emoncms or EmonPi you may need a URL of <http://x.x.x.x/emoncms>."
```
Comment (klaasnicolaas) [inline source/_integrations/emoncms.markdown]: The problem with textlint lies here, in line 34

Comment (borpin) [reply source/_integrations/emoncms.markdown]: But the lower case versions are in a URL.

**Comment by borpin** [comment]
@klaasnicolaas, thanks - I just followed what was there.

As this uses a template, is there a way to get VSCode to preview it correctly do you know?

File: source/_integrations/emoncms.markdown (hunk 2 above)
Comment (frenck) [reply source/_integrations/emoncms.markdown]: The problem here is quite simple, the last URL you made, isn't a valid URL.
Hence, not handled as one.

**Comment by frenck** [comment]
Adjusted PR to pass the build.

File: source/_integrations/emoncms.markdown [hunk 3]
```diff
@@ -21,17 +21,54 @@ sensor:
   id: 1
 ```
 
+As of Feb 2020 including the option `use_emoncms_unit` will discover all sensors from Emoncms and use the unit of measurement specified in the Feed in Emoncms. Tested with [Emoncms](https://github.com/emoncms/emoncms) V10.1.13.
```
Comment (frenck) [inline source/_integrations/emoncms.markdown]:
```suggestion
As of Feb 2020 including the option `use_emoncms_unit` will discover all sensors from Emoncms and use the unit of measurement specified in the feed from Emoncms. Tested with [Emoncms](https://github.com/emoncms/emoncms) V10.1.13.
```

File: source/_integrations/emoncms.markdown [hunk 4]
```diff
@@ -21,17 +21,54 @@ sensor:
...
-- `value_template` (*Optional*): Defines a [template](/docs/configuration/templating/#processing-incoming-data) to alter the feed value.
-- `scan_interval` (*Optional*): Defines the update interval of the sensor in seconds.
-- `unit_of_measurement` (*Optional*): Defines the unit of measurement of for all the sensors. default is "W".
+{% configuration %}
+api_key:
+  description: The read API key for your Emoncms user.
+  required: true
+  type: string
+url:
+  description: "The base URL of Emoncms, use <https://emoncms.org> for the cloud based version. For self hosted Emoncms or EmonPi you may need a URL of `http://x.x.x.x/emoncms`."
```
Comment (frenck) [inline source/_integrations/emoncms.markdown]:
```suggestion
  description: "The base URL of Emoncms, use <https://emoncms.org> for the cloud-based version. For self-hosted Emoncms or EmonPi you may need a URL of `http://x.x.x.x/emoncms`."
```

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/emoncms.markdown (hunk 2 above)
Comment (borpin) [reply source/_integrations/emoncms.markdown]: How would you suggest identifying the need for a private/LAN IP here then?

<http://<YOUR_IP>/emoncms>

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by klaasnicolaas — DISMISSED** [review]

**Comment by borpin** [comment]
Updated to match agreed parent PR

**Review by frenck — APPROVED** [review]

**Pull request merged**